src/engine/Server.cpp

```diff
@@ -762,6 +762,20 @@ CPP_template_def(typename RequestT)(
   return messageSender;
 }
 
+// _____________________________________________________________________________
+ad_utility::MediaType Server::chooseBestFittingMediaType(
+    const std::vector<ad_utility::MediaType>& candidates,
+    const PlannedQuery& plannedQuery) {
+  if (candidates.empty()) {
+    return plannedQuery.parsedQuery_.hasConstructClause()
+               ? MediaType::turtle
+               : MediaType::sparqlJson;
+  } else {
+    // Very simple approach for now.
+    return candidates.front();
+  }
+}
```

Comment:
Isn't this too simple?
e.g. if the first media type is not compatible with the query type, we should use the second one etc, and if none of them is compatible, then we can use your turtle/json fallback
(should be simple, just use a whitelist of the supported mediatypes for CONSTRUCT/SELECT